pygmt/helpers/utils.py

```diff
@@ -267,3 +269,48 @@ def args_in_kwargs(args, kwargs):
         If one of the required arguments is in ``kwargs``.
     """
     return any(arg in kwargs for arg in args)
+
+
+def return_table(result, data_format, format_parameter, df_columns):
```

Comment:
Looks promising! A few comments:
1. I think you will need to use this with one of the existing functions so that we can test it out in this PR. In my opinion, grdtrack would be a good option.
2. My preference would be to add a `format_options` parameter for `return_table`. This could take a list, with defaults values including numpy, pandas, and str. This way, if it doesn't for example make sense to return string values then that could not be given as an option in the individual function documentation/implementation.
3. table-like options for input are now `str or numpy.ndarray or pandas.DataFrame or xarray.Dataset or geopandas.GeoDataFrame`. It would be nice if all of these were output options too.
4. I would prefer a more description argument for requested data format than **a**|**d**|**s**. Something like `numpy`|`pandas`|`str` seems more readable.
5. I think `df_columns` needs to be optional.

Reply:
>     3. table-like options for input are now `str or numpy.ndarray or 
>     pandas.DataFrame or xarray.Dataset or geopandas.GeoDataFrame`. 
>     It would be nice if all of these were output options too.

Sounds good; I'll have to get smarter on the last two but I don't see why it should be a problem.

>     4. I would prefer a more description argument for requested data format than **a**|**d**|**s**. 
>     Something like `numpy`|`pandas`|`str` seems more readable.

I like the idea of keeping it short, especially when there is a default option (I anticipate it being a numpy array) and the strings are not also the same word as Python modules or variable types. But I understand how the single letters could be confusing.

> 
>     5. I think `df_columns` needs to be optional.

Since this is a helper function, I envisioned that the argument for `df_columns` would be set up in the GMT function that is using it, such as using `["x", "y", "z"]` when calling this function inside of `grd2xyz`.

Reply:
>     3. table-like options for input are now `str or numpy.ndarray or pandas.DataFrame or xarray.Dataset or geopandas.GeoDataFrame`. It would be nice if all of these were output options too.

Added in c8cef5e

Reply:
>> 4. I would prefer a more description argument for requested data format than **a**|**d**|**s**. 
    Something like `numpy`|`pandas`|`str` seems more readable.

> I like the idea of keeping it short, especially when there is a default option (I anticipate it being a numpy array) and the strings are not also the same word as Python modules or variable types. But I understand how the single letters could be confusing.

I'll have to agree with Meghan that long descriptive names like `numpy`|`pandas`|`str` are preferable :slightly_smiling_face: